### ISSUE TITLE
shell integration conflicts with nvs

### ISSUE THREAD

**Issue by sbatten** [microsoft/vscode#167764]
Steps to Reproduce:

Using pwsh preview on Windows

1. Install nvs (node version switcher) 
2. Add to $profile: `nvs auto on`
3. Enable shell integration
4. Open terminal in vs code repo

🐛 => extra stuff on prompt
<img width="935" alt="image" src="https://user-images.githubusercontent.com/6561887/204901341-b4a53be7-d9cb-4d7f-a1b4-0227da7fe3e5.png">

**Comment by Tyriar** [comment]
@sbatten I can't repro, either running nvs auto on manually or in profile, with and without the starship prompt script as well

**Comment by meganrogge** [comment]
are you using pwsh preview @Tyriar ?

**Comment by Tyriar** [comment]
Don't think so, PowerShell 7.3.0

**Comment by Tyriar** [comment]
@sbatten you sure this isn't nvs being naughty, not us?

**Comment by sbatten** [comment]
no I am not sure but it doesn't happen in windows terminal

**Comment by sbatten** [comment]
and it doesn't happen if I disable shell integration

**Comment by Tyriar** [comment]
I can repro with pwsh preview.

The original prompt is set to `. "C:\Users\Daniel\AppData\Local\nvs\nvs.ps1" "prompt"`, which when run prints something similar to what you're seeing, after returning the prompt:

![image](https://user-images.githubusercontent.com/2193314/204936571-4c19e6b0-04ba-4b6c-b889-490c080baa71.png)

We invoke it here:

https://github.com/microsoft/vscode/blob/968026a46db6ad769e0d9f3a048d2c7d780de083/src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1#L76

Did something change around how `Invoke()` ends in pwsh preview? That would make sense if the preview waits for everything until printing and stable does not. That's my theory anyway and this seems like an issue on nvs' end.

**Comment by sbatten** [comment]
I repro with stable 7.3.0 as well, not just preview. And only in VS Code's terminal with shell integration enabled.

**Comment by Tyriar** [comment]
@sbatten could this be related to different output channels in pwsh and we might be printing everything but normally the prompt function doesn't?

**Comment by sbatten** [comment]
possibly, heres the code from nvs that's run

```
	# Output needs to be redirected to Write-Host, because stdout is ignored by prompt.
	# Process a byte at a time so that output like progress bars is real-time.
	$startInfo = New-Object System.Diagnostics.ProcessStartInfo $bootstrapNodePath
	$startInfo.Arguments = ($mainScript, "auto", "at", $pwd.Path)
	$startInfo.UseShellExecute = $false
	$startInfo.RedirectStandardOutput = $true
	$proc = [System.Diagnostics.Process]::Start($startInfo)
	while (($b = $proc.StandardOutput.Read()) -ne -1) {
		Write-Host -NoNewline ([char]$b)
	}
	$proc.WaitForExit
	$exitCode = $proc.ExitCode
}

```

and this is type of the object that prints that extra info
```
 TypeName: System.Management.Automation.PSMethod`1[[System.Management.Automation.MethodGroup`2[[System.Func`1[[System.Management.Automation.VOID, System.Management.Automation, Version=7.3.0.101, Culture=neutral, PublicKeyToken=31bf3856ad364e35]],
System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.Management.Automation.MethodGroup`2[[System.Func`2[[System.Int32, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral,
PublicKeyToken=7cec85d7bea7798e],[System.Boolean, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], System.Private.CoreLib, Version=7.0.0.0, Culture=neutral,
PublicKeyToken=7cec85d7bea7798e],[System.Func`2[[System.TimeSpan, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.Boolean, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral,
PublicKeyToken=7cec85d7bea7798e]], System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], System.Management.Automation, Version=7.3.0.101, Culture=neutral, PublicKeyToken=31bf3856ad364e35]], System.Management.Automation,        
Version=7.3.0.101, Culture=neutral, PublicKeyToken=31bf3856ad364e35]]

```


and here's what happens if you pipe to out-null
```
pwsh> (nvs.ps1 "prompt") | Out-Null                                                                                                                                                                                                                                       
PATH -= $env:LOCALAPPDATA\nvs\default
PATH += $env:LOCALAPPDATA\nvs\node\16.14.2\x64

```

**Comment by Tyriar** [comment]
@TylerLeonhardt any ideas on what the shell integration script might be doing wrong? This is where we add the result of the original prompt to our new prompt `$Result`:

https://github.com/microsoft/vscode/blob/55d6b17e19e1e8d4f6040097383e61524930e3c4/src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1#L76

Should we be piping or not using += or something for this?

**Comment by sbatten** [comment]
PR for fix in nvs with oh-my-posh also resolves this issue https://github.com/jasongin/nvs/pull/254, looks like the bug here in nvs that contributed to the problem is the missing parentheses on WaitForExit. That's just outputting the type of WaitForExit and we let that go to the screen. So yes, probably just an nvs bug, but I wonder why we get that information and other terminals don't

**Comment by Tyriar** [comment]
Nice, I think the output channel thing is likely the case, but since oh-my-posh also experiences the issue I think it's safe for us to close this out in favor of https://github.com/jasongin/nvs/pull/254. Haven't seen this problem elsewhere yet.

**Comment by jasongin** [comment]
This issue is fixed in [nvs 1.7.0](https://github.com/jasongin/nvs/releases/tag/v1.7.0).
